### PR TITLE
Added input validation to modal

### PR DESCRIPTION
**Pull request by WeissSa**
## Pull Request Summary
See title

### Change types

- [x] Breaking change (new feature which messes with old one)

- [x] Adding tests

- [x] Front end change

To manually test use the JSON files I sent to discord as instructed.

### CODE REVIEW

File: front-end/src/components/TranscriptUploadModal.jsx [hunk 1]
```diff
@@ -9,22 +9,38 @@ class TranscriptUploadModal extends React.Component {
...
+  }
+
+  validateData(data) {
+    try {
+      const parsedData = JSON.parse(data);
+      return parsedData.data;
+    } catch (error) {
+      return false;
+    }
+  }
```
Comment (emilylwan) [inline front-end/src/components/TranscriptUploadModal.jsx:40]: Is this going to impact the rest of the formatting for backend? Like needing less parsing etc.

**Review by emilylwan — APPROVED** [review]
Just the comment on how it might impact backend. Rest looks good!

File: front-end/src/components/TranscriptUploadModal.jsx (hunk 1 above)
Comment (WeissSa) [reply front-end/src/components/TranscriptUploadModal.jsx:40]: It has no effect on BE.

**Review by KayleeLChan — APPROVED** [review]
lets go team looks good to me 💃 💃 💃

**Pull request merged**